### PR TITLE
Random clean up and bug fixes

### PR DESCRIPTION
**Pull request by antroseco**

### CODE REVIEW

File: codegen/generatable.py [hunk 1]
```diff
@@ -72,8 +72,7 @@ def ir_ref_without_type_annotation(self) -> str:
         assert self.ir_reg is not None
         if isinstance(self.type.definition, CharArrayDefinition):
             return f"@.str.{self.ir_reg}"
-        else:
-            return f"@.{self.ir_reg}"
```
Comment (DaveDuck321) [inline codegen/generatable.py:76]: Fine :imp: I'll stop doing this

File: graphene_parser.py [hunk 2]
```diff
@@ -517,6 +521,20 @@ def try_parse_fn_from_specialization(
                 resolved_return = self._program.resolve_type(
                     unresolved_return.produce_specialized_copy(specialization_map)
                 )
+            except SubstitutionFailure:
+                return None  # SFINAE
```
Comment (DaveDuck321) [inline graphene_parser.py:525]: Wait... How did this work before?

File: std/algorithms.c3 [hunk 3]
```diff
@@ -36,3 +36,22 @@ function [T] reverse : (buffer : Span<T>&) -> void = {
...
+        if span.data[mid] < elem {
+            lo = mid + 1;
+        }
+        else {
+            hi = mid;
+        }
+    }
+
+    return lo;
+}
```
Comment (DaveDuck321) [inline std/algorithms.c3:57]: Amazing!

**Review by DaveDuck321 — APPROVED** [review]
Very pretty :heart_eyes:. Our UFCS looks very sexy now

File: codegen/generatable.py (hunk 1 above)
Comment (antroseco) [reply codegen/generatable.py:76]: Hahaha, feel free to edit pylintrc

File: graphene_parser.py (hunk 2 above)
Comment (antroseco) [reply graphene_parser.py:525]: We were always catching `SubstitutionFailure`s (now lines 553-554). I had to split the try/catch such that we can attach the location information of the return type specifically.

**Pull request merged**